### ISSUE TITLE
Calculating auxiliary residuals in dlm package

### ISSUE THREAD

**Issue by k41m4n** [Figuera/ElastH#1]
It seems that you have developed some script to calculate auxiliary residuals in dlm package (I could see your question in KFAS repository). Are they part of this repository. If not, would you be so kind to share it?

**Comment by Figuera** [comment]
You can check this file: https://github.com/Figuera/ElastH/blob/master/R/choques.R

I think the above file is fairly straightforward to read. But if you have any question feel free to ask.

**Comment by k41m4n** [comment]
Thank you for your reply.  Indeed, this is an excellent implementation of the algorithms to calculate  auxiliary residuals provided by Durbin and Koopman in their book Time Series Analysis by State Space Methods.

**Comment by k41m4n** [comment]
I have reviewed again the code in choques.R to calculate the error variances matrices of the smoothed disturbances.  

In line 39 , should it be: vareta <- lapply(1:n, function(t) **jdlm$W[[t]]** - jdlm$W[[t]]%*%N[[t]]%*%jdlm$W[[t]]) instead of
vareta <- lapply(1:n, function(t) jdlm$W[[t]]%*%N[[t]]%*%jdlm$W[[t]]) , 

and then in line 46, should it be: vareps <- lapply(1:n, function(t) **jdlm$V[[t]]** - jdlm$V[[t]]%*%DD[[t]]%*%jdlm$V[[t]]) instead of vareps <- lapply(1:n, function(t) jdlm$V[[t]]%*%DD[[t]]%*%jdlm$V[[t]])  ?

Please see equations in (4.69) on page 96 in Time Series Analysis by State Space Methods of Durbin and Koopman.

**Comment by Figuera** [comment]
@k41m4n 

Looks like you are correct. I must say that the `dlm` based code on my package is unmaintained as I transitioned away to use `KFAS` which makes it harder for me verify, in practice, if the modification you propose is desirable.

I am pretty sure that I was aware of equation 4.69 when i wrote the code which makes me wonder what lead to that mistake, I was using a commercial software as a benchmark at the time and maybe I was trying to match its results, or I simply wrote the wrong code and my testing never pointed to any problems.

For your purposes, I think you will be fine using the code you propose.
